frontend/src/pages/projects/screens/detail/connections/ManageConnectionsModal.tsx

```diff
@@ -0,0 +1,177 @@
+    ConnectionTypeConfigMapObj | undefined
+  >(enabledConnectionTypes?.length === 1 ? enabledConnectionTypes[0] : undefined);
+  const { data: nameDescData, onDataChange: setNameDescData } = useK8sNameDescriptionFieldData();
+  const [connectionValues, setConnectionValues] = React.useState<{
+    [key: string]: ConnectionTypeValueType;
+  }>(() => {
+    if (connection?.data) {
+      return connection.data;
+    }
+    if (enabledConnectionTypes?.length === 1) {
+      return getDefaultValues(enabledConnectionTypes[0]);
+    }
+    return {};
+  });
+
+  // if user changes connection types, don't discard previous entries in case of accident
+  const [previousEntries, setPreviousEntries] = React.useState<{
+    [connectionTypeName: string]: {
+      nameDesc: K8sNameDescriptionFieldData;
+      values: {
+        [key: string]: ConnectionTypeValueType;
+      };
+    };
+  }>({});
+  const changeSelectionType = React.useCallback(
+    (type?: ConnectionTypeConfigMapObj) => {
+      // save previous connection values
+      if (selectedConnectionType) {
+        setPreviousEntries({
+          ...previousEntries,
+          [selectedConnectionType.metadata.name]: {
+            nameDesc: nameDescData,
+            values: connectionValues,
+          },
+        });
+        // clear previous values
+        setNameDescData('name', '');
+        setNameDescData('description', '');
+        setConnectionValues({});
+      }
+      // load saved values?
+      if (type?.metadata.name && type.metadata.name in previousEntries) {
+        setNameDescData('name', previousEntries[type.metadata.name].nameDesc.name);
+        setNameDescData('description', previousEntries[type.metadata.name].nameDesc.description);
+        setConnectionValues(previousEntries[type.metadata.name].values);
+      } else {
+        // first time load, so add default values
+        setConnectionValues(getDefaultValues(type));
+      }
+
+      setSelectedConnectionType(type);
+    },
+    [selectedConnectionType, previousEntries, nameDescData, connectionValues, setNameDescData],
+  );
+
+  const isValid = React.useMemo(
+    () =>
+      !!selectedConnectionType &&
+      !!nameDescData.name &&
+      !selectedConnectionType.data?.fields?.find(
+        (field) =>
+          isConnectionTypeDataField(field) &&
+          field.required &&
+          !connectionValues[field.envVar] &&
+          field.type !== ConnectionTypeFieldType.Boolean,
+      ),
+    [selectedConnectionType, nameDescData, connectionValues],
+  );
+
+  return (
+    <Modal
+      title="Add Connection"
+      isOpen
+      onClose={() => {
+        onClose();
+      }}
+      variant="medium"
+      footer={
+        <DashboardModalFooter
+          submitLabel="Create"
+          onCancel={onClose}
+          onSubmit={() => {
+            const connectionValuesAsStrings = Object.fromEntries(
+              Object.entries(connectionValues).map(([key, value]) => [key, String(value)]),
+            );
```

Comment:
We need to discuss how dropdown values should be serialized / deserialized. Especially for multi dropdown. A multi drop down item containing a comma will be serialized as a string in a way that loses the ability to parse back the value.